iib/workers/tasks/utils.py

```diff
@@ -701,6 +701,23 @@ def run_cmd(
     return response.stdout
 
 
+def terminate_process(proc, timeout=5):
+    """
+    Terminate given process. Fallback to SIGKILL when process is not terminated in given timeout.
+
+    :param subprocess.Popen proc: process to be terminated
+    :param int timeout: number of seconds to wait for terminating process
+    """
+    log.debug('Terminating process %s', proc)
+    proc.terminate()
+    try:
+        proc.communicate(timeout=timeout)
```

Comment:
Just a small comment, I was going to ask why did you use `proc.communicate()` here instead of `proc.wait()` and I see that's because of the possible deadlock when using PIPE for STDIN and STDOUT as mentioned [here](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait).


However I think it might be useful for maintainability to add a comment above explaining the use of `proc.communicate()` to avoid the deadlock, WDYT?

Reply:
@JAVGan Yeah, why not. Comment added.